system/lib/wasm_worker/library_wasm_worker.c

```diff
@@ -37,20 +40,25 @@ emscripten_wasm_worker_t emscripten_create_wasm_worker(void *stackLowestAddress,
 	assert(emscripten_stack_get_base() <= (uintptr_t)stackLowestAddress || emscripten_stack_get_end() >= (uintptr_t)stackLowestAddress + stackSize
 		&& "When creating a Wasm Worker, its stack should be located either in global data or on the heap, not on the calling thread's own stack!");
 
-	// The Worker's TLS area will be spliced off from the stack region.
-    // We expect TLS area to need to be at most 16 bytes aligned
+	// We expect TLS area to need to be at most 16 bytes aligned
 	assert(__builtin_wasm_tls_align() == 0 || 16 % __builtin_wasm_tls_align() == 0);
 
-#ifndef NDEBUG
-	uint32_t tlsSize = (__builtin_wasm_tls_size() + 15) & -16;
-	assert(stackSize > tlsSize);
-#endif
+	// The Worker's TLS area will be spliced off from the stack region, so the
+	// stack needs to be at least as large as the TLS region.
+	uint32_t tlsSize = ROUND_UP(__builtin_wasm_tls_size(), 16);
+	_emscripten_errf("emscripten_create_wasm_worker %lu %d %d", __builtin_wasm_tls_size(), tlsSize, stackSize);
```

Comment:
should this error be actually descriptive? or was it just for debugging this test?

Reply:
Oops, didn't mean to include that debug line.